onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -67,12 +67,14 @@
 #include "core/optimizer/transpose_optimizer/ort_transpose_optimizer.h"
 #include "core/optimizer/unsqueeze_elimination.h"
 #include "core/optimizer/identical_children_consolidation.h"
-#ifdef ENABLE_TRAINING
+#ifdef ENABLE_TRAINING_CORE
 #include "orttraining/core/optimizer/bitmask_dropout_replacement.h"
 #include "orttraining/core/optimizer/bias_softmax_dropout_fusion.h"
-#include "orttraining/core/optimizer/memory_optimizer.h"
 #include "orttraining/core/optimizer/sce_loss_grad_bias_fusion.h"
 #endif
+#ifdef ENABLE_TRAINING
+#include "orttraining/core/optimizer/memory_optimizer.h"
```

Comment:
you don't need the recompute feature in training_api build?

Reply:
No, current target models which will use training api will be very simple... this optimizer is required for advanced scenarios when we want to increase the batch size by saving some memory and recomputing the activations again when it is time for gradient computation.